interfaces/builtin/ofono.go

```diff
@@ -0,0 +1,332 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/release"
+)
+
+const ofonoPermanentSlotAppArmor = `
+# Description: Allow operating as the ofono service. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+
+# ofono puts ppp on top of the tun device
+/dev/net/tun rw,
+# to create ppp network interfaces
+capability net_admin,
+
+# To check present devices
+/run/udev/data/+usb:* r,
+/run/udev/data/+pci:* r,
+/run/udev/data/c* r,
+/run/udev/data/n* r,
+/sys/bus/usb/devices/ r,
+# FIXME snapd should be querying udev and adding the /sys and /run/udev accesses
+# that are assigned to the snap, but we are not there yet.
+/sys/bus/usb/devices/** r,
+
+# To get current seat, used to know user preferences like default SIM in
+# multi-SIM devices.
+/run/systemd/seats/{,*} r,
+
+# Access to modem ports
+# FIXME snapd should be more dynamic to avoid conflicts between snaps trying to
+# access same ports.
+/dev/tty[^0-9]* rw,
+/dev/cdc-* rw,
+
+network netlink raw,
+network netlink dgram,
+network bridge,
+network inet,
+network inet6,
+network packet,
+network bluetooth,
+
+include <abstractions/nameservice>
+
+# DBus accesses
+include <abstractions/dbus-strict>
+
+dbus (send)
+    bus=system
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member={Request,Release}Name
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+    bus=system
+    name="org.ofono",
+
+# Allow traffic to/from our path and interface with any method
+dbus (receive, send)
+    bus=system
+    path=/{,**}
+    interface=org.ofono.*,
```

Comment:
This will allow everything to connect to ofono. Please use this instead:
```
# Allow traffic to/from our path and interface with any method for unconfined
# clients to talk to our ofono services.
dbus (receive, send)
    bus=system
    path=/{,**}
    interface=org.ofono.*
    peer=(label=unconfined),
```